internal/backend/db/db.go

```diff
@@ -135,6 +135,7 @@ type DB interface {
 	GetSessionLog(ctx context.Context, sessionID sdktypes.SessionID) (sdktypes.SessionLog, error)
 	UpdateSessionState(ctx context.Context, sessionID sdktypes.SessionID, state sdktypes.SessionState) error
 	AddSessionPrint(ctx context.Context, sessionID sdktypes.SessionID, print string) error
+	AddSessionStopRequested(ctx context.Context, sessionID sdktypes.SessionID, reason string) error
```

Comment:
Just my 2 cents - "add session stop requested" is a very weird function name, for multiple reasons:

- Multiple verbs, without an obvious meaning
- "Requested" implies an event/state, but this is a function

Reply:
tbh it does its job. if you have a better name lmk :) I will change it to `AddSessionStopRequest` though.